optuna/storages/_journal/file.py

```diff
@@ -164,28 +164,40 @@ def __init__(self, file_path: str, lock_obj: Optional[JournalFileBaseLock] = Non
     def read_logs(self, log_number_from: int) -> List[Dict[str, Any]]:
         logs = []
         with open(self._file_path, "rb") as f:
+            # Maintain remaining_log_size to allow writing by another process while reading the log
+            remaining_log_size = os.stat(self._file_path).st_size
             log_number_start = 0
             if log_number_from in self._log_number_offset:
                 f.seek(self._log_number_offset[log_number_from])
                 log_number_start = log_number_from
+                remaining_log_size -= self._log_number_offset[log_number_from]
 
             last_decode_error = None
             for log_number, line in enumerate(f, start=log_number_start):
+                if log_number < log_number_from:
+                    continue
```

Comment:
Sorry, I've noticed that we need to put this if statement under L189. Otherwise, remaining_log_size will be wrong.

```python
for log_number, line in ...:
    ...
    if log_number + 1 not in ...:
         self._log_number_offset[log_number + 1] = ...
    if log_number < log_number_from:
         continue
     # Ensure that ...
```